cmd/agent/main.go

```diff
@@ -50,6 +50,13 @@ func run(cmd *cobra.Command, args []string) error {
 		snapshotURL = fmt.Sprintf(DefaultSnapshotURLFmt, entrypoint.ExternalSnapshotPort)
 	}
 
+	metricsServer := agent.NewMetricsServer(ambAgent.MetricsRelayHandler)
+	go func() {
+		if err := metricsServer.StartServer(ctx); err != nil {
+			dlog.Error(ctx, err)
+		}
+	}()
```

Comment:
Don't ever launch a goroutine that you don't have a way to wait for it to shut down.  github.com/datawire/dlib/dgroup can help with this, but you are free to use other solutions as well.